piff/optical_model.py

```diff
@@ -155,12 +223,37 @@ def __init__(self, template=None, logger=None, **kwargs):
             raise TypeError("Required keyword argument 'diam' not found")
         if 'lam' not in self.opt_kwargs:
             raise TypeError("Required keyword argument 'lam' not found")
+        self.diam = self.opt_kwargs['diam']
+        self.lam = self.opt_kwargs['lam']
 
         # pupil_angle and strut_angle won't serialize properly, so repr them now in self.kwargs.
         for key in ['pupil_angle', 'strut_angle']:
             if key in self.kwargs:
                 self.kwargs[key] = repr(self.kwargs[key])
 
+        # build the Galsim optical aperture here to cache it
+        if  'pupil_plane_im' in self.opt_kwargs:
+            self.aperture = galsim.Aperture(diam=self.opt_kwargs['diam'],pupil_plane_im=self.opt_kwargs['pupil_plane_im'],
+                                            gsparams=self.gsparams)
+        else:
+            self.aperture = galsim.Aperture(**self.opt_kwargs)
+            #self.aperture = galsim.Aperture(diam=self.opt_kwargs['diam'], obscuration=self.opt_kwargs['obscuration'],
+            #                                nstruts=self.opt_kwargs['nstruts'], strut_thick=self.opt_kwargs['strut_thick'],
+            #                                strut_angle=self.opt_kwargs['strut_angle'],
+            #                                pupil_plane_scale=None, pupil_plane_size=None,
+            #                                oversampling=self.opt_kwargs['oversampling'], pad_factor=self.opt_kwargs['pad_factor'],
+            #                                gsparams=self.gsparams)
```

Comment:
Should delete these commented out lines.

Reply:
done